### PR TITLE
Added explicit configuration for the upload to S3

### PR DESCRIPTION
**Pull request by giocalitri**
#### What are the relevant tickets?
N/A but probably of #18

#### What's this PR do?
Exposes configuration for upload to S3 so we can tune it in the future

#### How should this be manually tested?
The upload to S3 should be slightly faster using the local environment, but functionally it should simply work as before.

### CODE REVIEW

**Comment by noisecapella** [comment]
Can you rebase please?

**Comment by giocalitri** [comment]
Done

File: odl_video/settings.py [hunk 1]
```diff
@@ -182,3 +182,27 @@
...
 VIDEO_S3_TRANSCODE_BUCKET = os.environ.get('VIDEO_S3_TRANSCODE_BUCKET', '{}_transcoded'.format(VIDEO_S3_BUCKET))
 VIDEO_S3_THUMBNAIL_BUCKET = os.environ.get('VIDEO_S3_THUMBNAIL_BUCKET', '{}_thumbnails'.format(VIDEO_S3_BUCKET))
+
+# AWS S3 upload settings
+# the defaults values come from the default configuration in boto3.s3.transfer.TransferConfig
+# apart from the first 2
+KB = 1024
+MB = KB * KB
+
+AWS_S3_UPLOAD_MULTIPART_THRESHOLD_MB = os.environ.get('AWS_S3_UPLOAD_MULTIPART_THRESHOLD_MB', 32)
```
Comment (noisecapella) [inline odl_video/settings.py]: These environment variables will be interpreted as strings. Can you parse these into ints here and below, and into a bool at the end? Maybe we should use `get_var` from the cookiecutter here

File: odl_video/settings.py [hunk 2]
```diff
@@ -182,3 +182,27 @@
...
+AWS_S3_UPLOAD_MULTIPART_CHUNKSIZE_MB = os.environ.get('AWS_S3_UPLOAD_MULTIPART_CHUNKSIZE_MB', 32)
+AWS_S3_UPLOAD_MAX_CONCURRENCY = os.environ.get('AWS_S3_UPLOAD_MAX_CONCURRENCY', 10)
+AWS_S3_UPLOAD_NUM_DOWNLOAD_ATTEMPTS = os.environ.get('AWS_S3_UPLOAD_MULTIPART_CHUNKSIZE_MB', 5)
```
Comment (noisecapella) [inline odl_video/settings.py]: The variable name and environment variable name don't match

File: odl_video/settings.py [hunk 3]
```diff
@@ -182,3 +182,27 @@
 VIDEO_S3_BUCKET = os.environ.get('VIDEO_S3_BUCKET', 'odl-video-service')
 VIDEO_S3_TRANSCODE_BUCKET = os.environ.get('VIDEO_S3_TRANSCODE_BUCKET', '{}_transcoded'.format(VIDEO_S3_BUCKET))
 VIDEO_S3_THUMBNAIL_BUCKET = os.environ.get('VIDEO_S3_THUMBNAIL_BUCKET', '{}_thumbnails'.format(VIDEO_S3_BUCKET))
+
+# AWS S3 upload settings
+# the defaults values come from the default configuration in boto3.s3.transfer.TransferConfig
+# apart from the first 2
```
Comment (noisecapella) [inline odl_video/settings.py]: Why are the first two different?

Comment (giocalitri) [inline odl_video/settings.py]: I made some tests and the default value of 8MB seems to not be optimal for the upload.

File: odl_video/settings.py (hunk 1 above)
Comment (giocalitri) [inline odl_video/settings.py]: sure

File: odl_video/settings.py (hunk 2 above)
Comment (giocalitri) [inline odl_video/settings.py]: copy/paste error

**Comment by giocalitri** [comment]
changes done

**Pull request merged**